kobuki_node/CMakeLists.txt

```diff
@@ -1,39 +1,100 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(kobuki_node)
-find_package(catkin REQUIRED COMPONENTS rospy roscpp nodelet pluginlib tf angles
-                                        geometry_msgs sensor_msgs nav_msgs std_msgs diagnostic_updater diagnostic_msgs
-                                        kobuki_msgs kobuki_driver kobuki_keyop kobuki_safety_controller
-                                        ecl_exceptions ecl_sigslots ecl_streams ecl_threads)
-
-catkin_package(
-   INCLUDE_DIRS include
-   LIBRARIES kobuki_ros kobuki_nodelet
-   CATKIN_DEPENDS rospy roscpp nodelet pluginlib tf angles
-                  geometry_msgs sensor_msgs nav_msgs std_msgs diagnostic_updater diagnostic_msgs
-                  kobuki_msgs kobuki_driver kobuki_keyop kobuki_safety_controller
-                  ecl_exceptions ecl_sigslots ecl_streams ecl_threads
-)
 
-include_directories(include ${catkin_INCLUDE_DIRS})
+find_package(ament_cmake_ros REQUIRED)
+find_package(angles REQUIRED)
+find_package(diagnostic_msgs REQUIRED)
+find_package(diagnostic_updater REQUIRED)
+find_package(ecl_build REQUIRED)
+find_package(ecl_errors REQUIRED)
+find_package(ecl_exceptions REQUIRED)
+find_package(ecl_geometry REQUIRED)
+find_package(ecl_linear_algebra REQUIRED)
+find_package(ecl_sigslots REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(kobuki_driver REQUIRED)
+find_package(kobuki_ros_interfaces REQUIRED)
+find_package(nav_msgs REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rclcpp_components REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(std_msgs REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(tf2_ros REQUIRED)
+
+ecl_enable_cxx14_compiler()
+ecl_enable_cxx_warnings()
 
-add_subdirectory(src)
+include_directories(include)
 
-install(DIRECTORY include/${PROJECT_NAME}/
-        DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+add_library(kobuki_ros SHARED
+  src/diagnostics.cpp
+  src/kobuki_ros.cpp
+  src/odometry.cpp
 )
-install(DIRECTORY plugins
-        DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+ament_target_dependencies(kobuki_ros
+  "angles"
+  "diagnostic_msgs"
+  "diagnostic_updater"
+  "ecl_errors"
+  "ecl_exceptions"
+  "ecl_geometry"
+  "ecl_linear_algebra"
+  "ecl_sigslots"
```

Comment:
Do these actually work? I suspect they're only accidentally getting picked up via another dependency. Ostensibly, they should all follow the `<package_name>::<package_name>` style with `target_link_libraries`.

Reply:
No, they don't.  And further, I have to do recursive dependencies through kobuki_driver.  I've fixed this up for now with some TODO comments in 9605787.